WeakAuras/WeakAuras.lua

```diff
@@ -1642,9 +1642,127 @@ do
   end
 end
 
+local recurseDiff
+function recurseDiff(ours, theirs, ignored)
+  for key, ourVal in pairs(ours) do
+    local ignoredVal = type(ignored) == 'table' and ignored[key]
+    if ignoredVal ~= true then
+      if type(ourVal) == 'table' then
+        local theirVal = theirs[key]
+        if type(theirs[key]) ~= 'table' 
+        or recurseDiff(ourVal, theirVal, ignoredVal or nil) then
```

Comment:
What if I wanted to ignore that an animation is used or not, but not ignore the contents of said animation? I would need to have a deeper ignoredTable. This is groundwork for further iteration.

Reply:
Well, don't try to guess where the next iteration will take you. That's premature generalization that often doesn't work out anyway. Case in point, you passed the wrong variable and as I pointed out, even if you passed on the right variable, it would have the wrong semantics.